mlocal/frags/build_cli.mk

```diff
@@ -8,11 +8,15 @@ $(singularity_build_config): $(BUILDDIR)/config.h $(SOURCEDIR)/scripts/go-genera
 
 CLEANFILES += $(singularity_build_config)
 
-# singularity
-singularity_SOURCE := $(shell $(SOURCEDIR)/makeit/gengodep -v2 "$(GO)" "$(SOURCEDIR)" "$(GO_TAGS)" "$(SOURCEDIR)/cmd/singularity")
+# contain singularity_SOURCE variable list
+singularity_deps := $(BUILDDIR_ABSPATH)/singularity.d
+
+$(singularity_deps): $(GO_MODFILES) $(singularity_SOURCE)
+	@echo " GEN GO DEP" $@
+	$(V)$(SOURCEDIR)/makeit/gengodep -v3 "$(GO)" "singularity_SOURCE" "$(GO_TAGS)" "$@" "$(SOURCEDIR)/cmd/singularity"
 
 singularity := $(BUILDDIR)/singularity
-$(singularity): $(singularity_build_config) $(singularity_SOURCE)
+$(singularity): $(singularity_build_config) $(singularity_deps) $(singularity_SOURCE)
```

Comment:
not sure why the singularity binary should depend on the file with the dependencies. Depending on the actual files listed in `singularity_SOURCE` should be enough, shouldn't it?

If something is being fixed by adding this dependency, there should be a comment about it, as it's not obvious what that might be.

Reply:
it's required to track module update to trigger the build, I could put `$(GO_MODFILES)` instead to have the same effect